src/pke/lib/scheme/ckksrns/ckksrns-leveledshe.cpp

```diff
@@ -360,15 +360,52 @@ std::vector<DCRTPoly::Integer> LeveledSHECKKSRNS::GetElementForEvalAddOrSub(
   const auto cryptoParams =
       std::static_pointer_cast<CryptoParametersCKKSRNS>(
           ciphertext->GetCryptoParameters());
-  double scFactor =
-      cryptoParams->GetScalingFactorReal(ciphertext->GetLevel());
 
-  DCRTPoly::Integer intScFactor = static_cast<uint64_t>(scFactor + 0.5);
+  double scFactor;
+  if (cryptoParams->GetRescalingTechnique() == FLEXIBLEAUTOEXT && ciphertext->GetLevel() == 0) {
+    scFactor = cryptoParams->GetScalingFactorRealBig(ciphertext->GetLevel());
+  } else {
+    scFactor = cryptoParams->GetScalingFactorReal(ciphertext->GetLevel());
+  }
+
+  // Compute approxFactor, a value to scale down by, in case the value exceeds a 64-bit integer.
+  int32_t MAX_BITS_IN_WORD = 62;
```

Comment:
is it possible to define MAX_BITS_IN_WORD  in one place?

Reply:
Defined it in constants.h